pkg/vsphere/compute/placement/placement_policy.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2018 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package placement
+
+import (
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/pkg/vsphere/performance"
+	"github.com/vmware/vic/pkg/vsphere/vm"
+)
+
+type rankedHost struct {
+	HostReference string
+	*performance.HostMetricsInfo
+	score float64
+}
+
+type rankedHosts []rankedHost
+
+func (r rankedHosts) Len() int           { return len(r) }
+func (r rankedHosts) Swap(i, j int)      { r[i], r[j] = r[j], r[i] }
+func (r rankedHosts) Less(i, j int) bool { return r[i].score > r[j].score }
+
+// HostPlacementPolicy defines the interface for using metrics to decide the appropriate host
+// for a VM based on host metrics and VM provisioned resources.
+type HostPlacementPolicy interface {
```

Comment:
Consider splitting this interface into a file called `placement.go` and the ranked definitions into a `ranked.go` along with the ranking logic.

Reply:
You read my mind. :) I've created `random.go`, `ranked.go` and `ranked_test.go`, and renamed `placement_policy.go` -> `placement.go`.